src/components/Disputes/DisputeAutoReveal.js

```diff
@@ -0,0 +1,119 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { Button, GU, Info } from '@aragon/ui'
+import { useWallet } from '../../providers/Wallet'
+
+import { getAutoRevealRequest } from '../../services/autoReveal'
+import {
+  getAutoRevealPreference,
+  getCodeFromLocalStorage,
+  getOutcomeFromCommitment,
+} from '../../utils/crvoting-utils'
+
+function DisputeAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+
+  const autoRevealRequested = useAutoRevealPolling(account, disputeId, roundId)
+
+  if (autoRevealRequested) {
+    return <Info>Auto reveal enabled!</Info>
+  }
+
+  return (
+    <RequestAutoReveal
+      commitment={commitment}
+      disputeId={disputeId}
+      onAutoReveal={onAutoReveal}
+      roundId={roundId}
+    />
+  )
+}
+
+function RequestAutoReveal({ commitment, disputeId, onAutoReveal, roundId }) {
+  const { account } = useWallet()
+  const autoRevealServiceEnabled = getAutoRevealPreference(account)
+
+  const handleSubmit = useCallback(
+    event => {
+      event.preventDefault()
+
+      const password = getCodeFromLocalStorage(account, disputeId)
+      const outcome = getOutcomeFromCommitment(commitment, password)
+
+      onAutoReveal(account, disputeId, roundId, outcome, password)
+    },
+    [account, commitment, disputeId, onAutoReveal, roundId]
+  )
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <Button
+        label="Reveal your vote with auto-reveal service"
+        mode="strong"
+        type="submit"
+        wide
+      />
+      {autoRevealServiceEnabled && (
+        <Info
+          mode="warning"
+          css={`
+            margin-top: ${2 * GU}px;
+          `}
+        >
+          The auto-reveal service request has failed. Please click on the button
+          above to resend the request, reveal your vote and enable the service.
+        </Info>
+      )}
+    </form>
+  )
+}
+
+function useAutoRevealPolling(account, disputeId, roundId) {
```

Comment:
Rather than polling, I wonder if we should only do this once on the component render?

We only use the polling to know if the user's request to re-register was successful, right?

Reply:
Yes exactly. When implementing this, that was the first direction i took, but given that the request is asynchronous (we use the request queue), didn't find a way to force re-running the effect.

Do you have one in mind?

Reply:
Would it be possible to use `await` on the `onAutoReveal()` in `handleSubmit()` and update state afterwards? Outside of a separate tab, you have to go through the `handleSubmit()` handler to change your auto reveal state, right?

Reply:
I don't think it would be possible because we are using the same flow as we do for normal transactions. This means that the `onAutoReveal()` would return when the auto reveal request is added to the request queue for processing and we don't have any more the control of that request.

Nevertheless, note that we stop polling if the auto reveal was requested successfully so in that case would only be doing a single query. And in the case that the juror didn't request the auto reveal we would be polling every 3 seconds which i don't think it's that bad.  

Do you see a potential issue that could arise?

Reply:
> Do you see a potential issue that could arise?

Nope, but just feels a bit awkward to poll when we _know_ when to check. But it is a nice side effect to cover the case that you updated the API in a separate tab / etc. (although this is honestly unlikely here—it's much more common for authentication mechanisms to do this type of polling).

Let's leave it like this for now, and add a few comments explaining why we're polling with `useAutoRevealPolling()`.